Chameleon-usage-examples/src/main/java/org/hibernate/omm/example/ProgrammaticClientSettingsCustomize.java

```diff
@@ -0,0 +1,33 @@
+package org.hibernate.omm.example;
+
+import org.hibernate.cfg.AvailableSettings;
+import org.hibernate.cfg.Configuration;
+import org.hibernate.engine.spi.SessionFactoryImplementor;
+import org.hibernate.omm.service.MongoClientSettingsCustomizer;
+
+public class ProgrammaticClientSettingsCustomize {
+
+  public static void main(String[] args) {
+
+    final var cfg = new Configuration();
+    cfg.setProperty(AvailableSettings.JAKARTA_JDBC_URL, "mongodb://127.0.0.1:27017/test");
+
+    // add your entity classes
+
+    final var standardServiceRegistryBuilder = cfg.getStandardServiceRegistryBuilder();
+
+    MongoClientSettingsCustomizer customizer = builder -> {
+      System.out.println("Do whatever you want");
+    };
```

Comment:
the only difference with your approach is simply to use Service rather than tweaking setting configuration.

Given we create such service instance purely in code, definitively we won't have the limit of no-arg constructor limit.

So we have multiple ways to go about the customization:

- customizer class name property (with limit of no-arg constructor)
- customizer spi service (with limit of no-arg constructor)
- complete programmatic way as show in this PR (I didn't include this in our tech doc, but we might need that ).

In any way, we need to change the logic on how to get and use the service (in this case it is MongoConnectionProvider class).